src/diffpy/utils/diffraction_objects.py

```diff
@@ -180,30 +180,32 @@ def __add__(self, other):
 
         Parameters
         ----------
-        other : DiffractionObject or int or float
-            The object to add to the current DiffractionObject. If `other` is a scalar value,
-            it will be added to all yarray. The length of the yarray must match if `other` is
-            an instance of DiffractionObject.
+        other : DiffractionObject, int, or float
+            The item to be added. If `other` is a scalar value, this value will be added to each element of the
+            yarray of this DiffractionObject instance. If `other` is another DiffractionObject, the yarrays of the
+            two DiffractionObjects will be combined element-wise. The result is a new DiffractionObject instance,
+            representing the addition and using the xarray from the left-hand side DiffractionObject.
 
         Returns
         -------
         DiffractionObject
-            The new and deep-copied DiffractionObject instance after adding values to the yarray.
+            THe new DiffractionObject instance with modified yarray values. This instance is a deep copy of the
+            original with the additions applied.
 
         Raises
         ------
         ValueError
-            Raised when the length of the yarray of the two DiffractionObject instances do not match.
+            Raised when the xarrays of two DiffractionObject instances are not equal.
```

Comment:
"not equal" feels more explicit and program-like, than "unequal" or "do not match"